packages/client/src/lib.js

```diff
@@ -180,13 +180,7 @@ class Web3Storage {
       headers: Web3Storage.headers(token)
     })
     if (!res.ok) {
-      // TODO: I'm assuming that an error for "CID isn't there (yet)" would be unergonomic. Need to verify.
-      // I'm thinking null means, nope, not yet, no can has. Anything else is _AN ERROR_
-      if (res.status === 404) {
-        return null
-      } else {
-        throw new Error(`${res.status} ${res.statusText}`)
-      }
+      throw new Error(`${res.status} ${res.statusText}`)
```

Comment:
Ah I didn't realise it threw here for non-2xx status. This means everyone has to `try`/`catch` _as well as_ inspect `res.ok`. Conversely, `fetch` **never** throws so you only ever have to check `res.ok`. I vote we match that behaviour.

```suggestion
```

Reply:
I agree with @alanshaw 's point